aten/src/ATen/native/DilatedConvolution.cpp

```diff
@@ -0,0 +1,650 @@
+    const IntArrayRef input_size,
+    const IntArrayRef output_size,
+    const IntArrayRef kernel_size,
+    const IntArrayRef stride_size,
+    const IntArrayRef pad_size,
+    const IntArrayRef dilation_size,
+    Dtype* data_col) {
+  if (dim == 3) {
+    vol2col<Dtype>(
+        data_hvol,
+        channels,
+        input_size[0],
+        input_size[1],
+        input_size[2],
+        output_size[0],
+        output_size[1],
+        output_size[2],
+        kernel_size[0],
+        kernel_size[1],
+        kernel_size[2],
+        pad_size[0],
+        pad_size[1],
+        pad_size[2],
+        stride_size[0],
+        stride_size[1],
+        stride_size[2],
+        dilation_size[0],
+        dilation_size[1],
+        dilation_size[2],
+        data_col);
+  }
+  if (dim == 2) {
+    im2col<Dtype>(
+        data_hvol,
+        channels,
+        input_size[0],
+        input_size[1],
+        output_size[0],
+        output_size[1],
+        kernel_size[0],
+        kernel_size[1],
+        pad_size[0],
+        pad_size[1],
+        stride_size[0],
+        stride_size[1],
+        dilation_size[0],
+        dilation_size[1],
+        data_col);
+  }
+}
+
+// column to hyper-volume, CPU
+template <typename Dtype, int64_t dim>
+void col2hvol(
+    const Dtype* data_col,
+    const int channels,
+    const IntArrayRef input_size,
+    const IntArrayRef output_size,
+    const IntArrayRef kernel_size,
+    const IntArrayRef stride_size,
+    const IntArrayRef pad_size,
+    const IntArrayRef dilation_size,
+    Dtype* data_hvol) {
+  if (dim == 3) {
+    col2vol<Dtype>(
+        data_col,
+        channels,
+        input_size[0],
+        input_size[1],
+        input_size[2],
+        output_size[0],
+        output_size[1],
+        output_size[2],
+        kernel_size[0],
+        kernel_size[1],
+        kernel_size[2],
+        pad_size[0],
+        pad_size[1],
+        pad_size[2],
+        stride_size[0],
+        stride_size[1],
+        stride_size[2],
+        dilation_size[0],
+        dilation_size[1],
+        dilation_size[2],
+        data_hvol);
+  }
+  if (dim == 2) {
+    col2im<Dtype>(
+        data_col,
+        channels,
+        input_size[0],
+        input_size[1],
+        output_size[0],
+        output_size[1],
+        kernel_size[0],
+        kernel_size[1],
+        pad_size[0],
+        pad_size[1],
+        stride_size[0],
+        stride_size[1],
+        dilation_size[0],
+        dilation_size[1],
+        data_hvol);
+  }
+}
+
+/*
+   check tensor data locations
+*/
+void conv_dilated_location_check(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    const Tensor& grad_output) {
+  // checking data locations of user-provided tensor arguments
+  TensorArg input_arg{input, "input", 2}, weight_arg{weight, "weight", 3},
+      bias_arg{bias, "bias", 4}, grad_output_arg{grad_output, "grad_output", 5};
+
+  // TODO: how to check that all tensors data is on CPU memory?
```

Comment:
`checkBackend` with `Backend::CPU`, I suppose. Or you could add the needed helper to TensorUtils.h (do that in another PR though lol)

Reply:
Thanks for the hint! Fixed.